WordPress/Classes/ViewRelated/Post/PostEditorState.swift

```diff
@@ -186,41 +176,68 @@ public class PostEditorStateContext {
         self.delegate = delegate
 
         guard let originalPostStatus = originalPostStatus else {
-            editorState = PostEditorStatePublish()
+            action = .publish
             return
         }
 
         switch originalPostStatus {
         case .draft where userCanPublish == false:
-            editorState = PostEditorStateSubmitForReview()
+            action = .submitForReview
         default:
-            editorState = PostEditorStateUpdate()
+            action = .update
+
         }
     }
 
     /// Call when the post status has changed due to a remote operation
     ///
     func updated(postStatus: BasePost.Status) {
         currentPostStatus = postStatus
-        let updatedState = editorState.updated(postStatus: postStatus, context: self)
-        guard type(of: editorState) != type(of: updatedState) else {
-            return
-        }
 
-        editorState = updatedState
+        let updatedAction = { () -> PostEditorAction in
+            switch postStatus {
+            case .draft where originalPostStatus != nil && originalPostStatus != .draft:
+                return .update
+            case .draft, .pending:
+                return .save
+            case .publish where originalPostStatus == nil || originalPostStatus == .draft:
```

Comment:
I think it's only a small amount of redundancy in those areas, and it may well be good redundancy as it makes it more explicit / easier to follow in each case.